consonance-webservice/src/main/java/io/swagger/workflow/api/impl/Ga4ghApiServiceImpl.java

```diff
@@ -0,0 +1,239 @@
+import io.swagger.workflow.model.Ga4ghWesWorkflowListResponse;
+import io.swagger.workflow.model.Ga4ghWesWorkflowLog;
+import io.swagger.workflow.model.Ga4ghWesWorkflowRequest;
+import io.swagger.workflow.model.Ga4ghWesWorkflowRunId;
+import io.swagger.workflow.model.Ga4ghWesWorkflowStatus;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+import io.swagger.workflow.api.NotFoundException;
+
+import java.io.InputStream;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpStatus;
+import org.glassfish.jersey.media.multipart.FormDataContentDisposition;
+
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import javax.validation.constraints.*;
+
+@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaJerseyServerCodegen", date = "2017-12-24T02:51:26.646Z")
+public class Ga4ghApiServiceImpl extends Ga4ghApiService {
+
+    private static ConsonanceWebserviceConfiguration config;
+    private static OrderResource orderResource;
+
+    public static void setConfig(ConsonanceWebserviceConfiguration config) {
+        Ga4ghApiServiceImpl.config = config;
+    }
+
+    public static void setOrderResource(OrderResource orderResource) {
+        Ga4ghApiServiceImpl.orderResource = orderResource;
+    }
+
+    @Override
+    public Response cancelJob(String workflowId, ConsonanceUser user) throws NotFoundException {
+        // do some magic!
+        return Response.ok().entity(new ApiResponseMessage(ApiResponseMessage.ERROR, "delete is not yet supported in Consonance!")).build();
+    }
+    @Override
+    public Response getServiceInfo(ConsonanceUser user) throws NotFoundException {
+        try {
+            Ga4ghWesServiceInfo serviceInfo = new Ga4ghWesServiceInfo();
+
+            // types
+            Ga4ghWesWorkflowTypeVersion wtv = new Ga4ghWesWorkflowTypeVersion();
+            wtv.addWorkflowTypeVersionItem("1.0.0");
+            Map<String, Ga4ghWesWorkflowTypeVersion> map = new HashMap<String, Ga4ghWesWorkflowTypeVersion>();
+            map.put("CWL", wtv);
+            map.put("WDL", wtv);
+            serviceInfo.setWorkflowTypeVersions(map);
+
+            // wes version
+            serviceInfo.setSupportedWesVersions(Arrays.asList("1.0.0"));
+
+            // engine
+            Map<String, String> engineMap = new HashMap<String, String>();
+            // FIXME: these need to be coordinated with the versions installed by the playbook for the worker
+            engineMap.put("Cromwell", "29");
+            engineMap.put("cwltool", "1.0.20170828135420");
+            serviceInfo.setEngineVersions(engineMap);
+
+            // FIXME: need to confirm this list with the Dockstore CLI
+            serviceInfo.setSupportedFilesystemProtocols(Arrays.asList("http", "https", "s3", "file"));
```

Comment:
Yeah, you can probably do some parsing of `dockstore plugin list` or create a PR for it to return something more machine-readible when running with something like `dockstore --script plugin list`

Reply:
There are also the default protocols provided via vfs2 https://github.com/ga4gh/dockstore/blob/3083a8769f9096a2112c2f175de538afd70c0ba3/dockstore-client/src/main/java/io/dockstore/common/FileProvisioning.java#L388

Reply:
Thanks Denis.  You, Charles, and I should chat about making the dockstore CLI more friendly for WES to be built on top of.  I have some ideas.